text/0000-programmatic-style-injection.md

```diff
@@ -127,11 +127,15 @@ import styles from "./styles.css" assert { type: "css" };
 console.log(styles + '') // [object CSSStyleSheet]
 ```
 
+An alternative to exporting LWC stylesheet module as an opaque object is to have a utility function `registerStylesheet` that registers compiled stylesheets so that component authors cannot modify or replace stylesheet functions. This would be a similar approach on component templates ([`isTemplateRegistered`](https://github.com/salesforce/lwc/blob/1b85b4606f8cc8a18c368d9ce3b9c3616043f10c/packages/@lwc/engine-core/src/framework/secure-template.ts#L17)).
+
 #### LWC stylesheet import enforcement
 
 As mentioned above, the LWC compiler implicitly loads the stylesheet associated with each template. Internally, the template compiler generates an import statement with the template file name and the `.css` extension. Since stylesheets are optional, the LWC module resolver currently ignores non-existent stylesheet imports.
 
-This behavior is problematic for explicit stylesheet imports as the compilation doesn't fail if the imported stylesheet can't be resolved. To solve this issue a new `optional` query string can be added to the import identifier to indicate whether or not the LWC compiler should report an error when the file is missing.
+This behavior is problematic for explicit stylesheet imports as the compilation doesn't fail if the imported stylesheet can't be resolved. To solve this issue, the LWC compiler should treat component's implict stylesheet import as optional and other explicit stylesheet imports as required. For DX improvement, LWC compiler should fail compilation or at least warn developers if an explicit stylesheet import cannot be resolved before treating non-existent stylesheet imports as empty imports. 
+
+An alternative solution is that a new `optional` query string can be added to the import identifier to indicate whether or not the LWC compiler should report an error when the file is missing.

```

Comment:
I don't think `?optional=true` is an alternative solution. Instead, I see it as an under-the-hood implementation detail of `To solve this issue, the LWC compiler should treat component's implict stylesheet import as optional and other explicit stylesheet imports as required.`. @rui-rayqiu